services/horizon/internal/httpx/router.go

```diff
@@ -166,7 +166,15 @@ func (r *Router) addRoutes(config *RouterConfig, rateLimiter *throttled.HTTPRate
 
 		r.Route("/liquidity_pools", func(r chi.Router) {
 			r.With(stateMiddleware.Wrap).Method(http.MethodGet, "/", restPageHandler(ledgerState, actions.GetLiquidityPoolsHandler{LedgerState: ledgerState}))
-			r.With(stateMiddleware.Wrap).Method(http.MethodGet, "/{id}", ObjectActionHandler{actions.GetLiquidityPoolByIDHandler{}})
+			r.Route("/{liquidity_pool_id}", func(r chi.Router) {
```

Comment:
To ensure `liquidity_pool_id` is not empty.
```suggestion
			r.Route("/{liquidity_pool_id:\\w+}", func(r chi.Router) {
```